src/EnergyPlus/HeatBalanceIntRadExchange.cc

```diff
@@ -247,142 +268,154 @@ namespace HeatBalanceIntRadExchange {
 						// For windows with an interior shade or blind an effective inside surface temp
 						// and emiss is used here that is a weighted combination of shade/blind and glass temp and emiss.
-					} else if ( surface_window.ShadingFlag == IntShadeOn || surface_window.ShadingFlag == IntBlindOn ) {
-						RecSurfTemp = surface_window.EffInsSurfTemp;
-						RecSurfEmiss = InterpSlatAng( surface_window.SlatAngThisTS, surface_window.MovableSlats, surface_window.EffShBlindEmiss ) + InterpSlatAng( surface_window.SlatAngThisTS, surface_window.MovableSlats, surface_window.EffGlassEmiss );
+						SurfaceTempK4[ ZoneSurfNum ] = window.EffInsSurfTemp;
+						SurfaceEmiss[ ZoneSurfNum ] = 
+							InterpSlatAng( window.SlatAngThisTS, window.MovableSlats, window.EffShBlindEmiss ) + 
+							InterpSlatAng( window.SlatAngThisTS, window.MovableSlats, window.EffGlassEmiss );					
 					} else {
-						RecSurfTemp = SurfaceTemp( RecSurfNum );
-						RecSurfEmiss = construct.InsideAbsorpThermal;
+						SurfaceTempK4[ ZoneSurfNum ] = SurfaceTemp( SurfNum );
+						SurfaceEmiss[ ZoneSurfNum ] = construct.InsideAbsorpThermal;
 					}
 				} else {
-					RecSurfTemp = SurfaceTemp( RecSurfNum );
-					RecSurfEmiss = construct.InsideAbsorpThermal;
+					SurfaceTempK4[ ZoneSurfNum ] = SurfaceTemp( SurfNum );
+					SurfaceEmiss[ ZoneSurfNum ] = construct.InsideAbsorpThermal;
 				}
-				// precalculate the fourth power of surface temperature as part of strategy to reduce calculation time - Glazer 2011-04-22
-				RecSurfTempInKTo4th = pow_4( RecSurfTemp + KelvinConv );
-				//      IF (ABS(RecSurfTempInKTo4th) > 1.d100) THEN
-				//        SendZoneSurfNum=0
-				//      ENDIF
+			} // for ZoneSurfNum
+
+			// Amir Roth 2015-07-01: Split off SurfaceTemp = pow4(SurfaceTemp) calculation so that it will vectorize.
+
+			// __restrict is a hint that tells the compiler that the array vecSurfaceTempK4 does not overlap in memory with 
+			// any other array, this helps the compiler decide that the code is vectorizable.
+			Real64 * __restrict vecSurfaceTempK4( &SurfaceTempK4[ 0 ] );
+			// __ep_assume_aligned tells the compiler that vecSurfaceTempK4 is aligned in memory allowing it to generate
+			// optimized vector load/store code. 
+			__ep_assume_aligned(Real64 *, vecSurfaceTempK4, 16);
+			// Both __restrict and __assume_aligned hints require raw pointers, hence need to use raw pointers here.  These 
+			// pointers do not manage or "own" memory, they simply point at memory in a way that allows the compiler to optimize
+			// better
+			for ( int ZoneSurfNum = 0; ZoneSurfNum < zvfi.NumOfSurfaces; ++ZoneSurfNum ) {
+				vecSurfaceTempK4[ ZoneSurfNum ] = pow_4( vecSurfaceTempK4 [ ZoneSurfNum ] + KelvinConv );
+			} // for ZoneSurfNum
+
+
+			// See comments above for explanation of __restrict and __ep_assume_aligned
+			Real64 * __restrict vecNetLWRadToSurf_Temp( &NetLWRadToSurf_Temp[ 0 ] );
```

Comment:
@amirroth Is this line needed? clang is griping about unused variables, and it looks like the same thing is happening down on line 373. Similar question for the three lines after this one.